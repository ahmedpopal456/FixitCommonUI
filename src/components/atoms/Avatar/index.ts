import { AvatarBadge } from './Avatar';
export { AvatarBadge};