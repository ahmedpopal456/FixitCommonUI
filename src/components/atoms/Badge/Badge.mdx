---
name: Badge
menu: Atoms
---
import { Props, Playground } from "docz";
import { Badge } from "./Badge.tsx";
import { DoczContainer } from '../../../utils/DoczContainer.tsx'

# Badge
The badge components dispays a number in a circle

# Usage
```tsx
import { Badge } from "fixit-common-ui";
<Badge>1</Badge>
```

## Props
<Props of={Badge}></Props>

## Default
<Playground>
  <DoczContainer>
    <Badge>1</Badge>
    <Badge>12</Badge>
    <Badge>12345</Badge>
  </DoczContainer>
</Playground>

## Custom
<Playground>
  <DoczContainer>
    <Badge background="red" color="light">12</Badge>
    <Badge background="accent" border="primary" color="primary">12</Badge>
  </DoczContainer>
</Playground>

## See through
<Playground>
  <DoczContainer>
    <Badge opacity={.2} color="accent" background="accent">12</Badge>
  </DoczContainer>
</Playground>
