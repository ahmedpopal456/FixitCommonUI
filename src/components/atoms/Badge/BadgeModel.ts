import colors from '../../../theme/colors';

export interface BadgeContainerProps{
  background?: keyof typeof colors;
  border?: keyof typeof colors;
  opacity?: number;
}

export interface BadgeProps extends BadgeContainerProps{
  color?: keyof typeof colors;
  children: string;
}
