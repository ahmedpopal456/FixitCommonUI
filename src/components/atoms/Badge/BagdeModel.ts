import colors from "../../../theme/colors";

  export interface BadgeProps {
      /** Can be any color from the theme color palette */
      color?: keyof typeof colors;
      /** Can be any color from the theme color palette */
      border?: keyof typeof colors;
      /** Can be any color from the theme color palette */
      background?: keyof typeof colors;
      /** Float between 0 and 1 */
      opacity?: number;
      /** Children which will be contained in the badge */
      children: string;
  }
