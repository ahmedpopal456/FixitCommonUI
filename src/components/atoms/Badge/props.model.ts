import colors from "../../../theme/colors";
  
  export interface Props {
    /**  Example of an optional prop */
    optional?: string;
    /** Example of a required prop*/
    required: boolean;
  }