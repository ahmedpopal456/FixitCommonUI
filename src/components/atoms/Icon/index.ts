import { Icon } from './Icon';
import { IconLibraries } from './IconLibraries';

export { Icon, IconLibraries };
