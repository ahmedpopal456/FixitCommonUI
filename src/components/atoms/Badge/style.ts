import styled from 'styled-components/native';
import { BadgeContainerProps } from './BadgeModel';
import colors from '../../../theme/colors';

// input needs ot be 6 char hex (not shortcut 3 char hex)
const hex2rgba = (hex: string, alpha = 1) => {
  const [r, g, b] = hex.match(/\w\w/g)!.map((x) => parseInt(x, 16));
  return `rgba(${r},${g},${b},${alpha})`;
};

/**
 * Styled Components
 *
 * Here you can add css to react native components to create a styled version of it
 */
export const BadgeContainer = styled.View`
    minWidth:40px;
    height:40px;
    border-radius:20px;
    border:2px solid;
    margin: 5px;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    align-self: flex-start;
    padding: 5px;
     ${(badgeContainerProps: BadgeContainerProps) => badgeContainerProps.background && `
          backgroundColor: ${colors[badgeContainerProps.background]};
          borderColor:${colors[badgeContainerProps.background]};
      `}
     ${(badgeContainerProps: BadgeContainerProps) => badgeContainerProps.border && `
          borderColor: ${colors[badgeContainerProps.border]};
      `}
     ${(badgeContainerProps: BadgeContainerProps) => badgeContainerProps.opacity && `
          backgroundColor: ${hex2rgba(colors[badgeContainerProps.background!], badgeContainerProps.opacity)};
      `}
`;
