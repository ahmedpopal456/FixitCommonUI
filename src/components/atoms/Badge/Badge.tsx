import React from 'react';
import { BadgeContainer } from './style';
import { BadgeProps } from './BadgeModel';
import { Label } from '../Typography';

/**
 * Describe the Component
 * @param props - The Badge's props
 * @returns A react native custom Badge component
 */
export const Badge: React.FC<BadgeProps> = (badgeProps:BadgeProps): JSX.Element => {
  const {
    opacity, background, border, children,
  } = badgeProps;
  return (
    <BadgeContainer opacity={opacity} background={background} border={border}>
      <Label justification="center" color={badgeProps.color} fontWeight={400} fontSize={20}>{children}</Label>
    </BadgeContainer>
  );
};

/** Setting the defaults for the props */
Badge.defaultProps = {
  color: 'accent',
  background: 'primary',
  opacity: 1,
};
