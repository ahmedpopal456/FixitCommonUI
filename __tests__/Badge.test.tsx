import 'react-native';
import React from 'react';
import { Badge } from '../src/components/atoms/Badge';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

/** Testing Rendered Output (snapshot testing) */
it('matches default snapshot', () => {
  const tag = renderer.create(<Badge>1</Badge>);
  expect(tag.toJSON()).toMatchSnapshot();
});

it('matches text color snapshot', () => {
  const tag = renderer.create(<Badge color="accent">1</Badge>);
  expect(tag.toJSON()).toMatchSnapshot();
});

it('matches background color snapshot', () => {
  const tag = renderer.create(<Badge background="accent" color="primary">1</Badge>);
  expect(tag.toJSON()).toMatchSnapshot();
});

it('matches border color snapshot', () => {
  const tag = renderer.create(<Badge background="accent" color="primary" border="red">1</Badge>);
  expect(tag.toJSON()).toMatchSnapshot();
});

it('matches opacity snapshot', () => {
  const tag = renderer.create(<Badge background="accent" color="primary" border="red" opacity={0.3}>1</Badge>);
  expect(tag.toJSON()).toMatchSnapshot();
});
