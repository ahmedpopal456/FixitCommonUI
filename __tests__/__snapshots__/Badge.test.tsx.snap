// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches background color snapshot 1`] = `
<View
  background="accent"
  opacity={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "rgba(255,209,74,1)",
        "borderColor": "#FFD14A",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "minWidth": 40,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="primary"
      fontSize={20}
      fontWeight={400}
      justification="center"
      padding={3}
      style={
        Array [
          Object {
            "color": "#33343E",
            "flexDirection": "row",
            "fontFamily": "Raleway",
            "fontSize": 20,
            "fontWeight": "400",
            "paddingBottom": 3,
            "paddingLeft": 3,
            "paddingRight": 3,
            "paddingTop": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`matches border color snapshot 1`] = `
<View
  background="accent"
  border="red"
  opacity={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "rgba(255,209,74,1)",
        "borderColor": "#D4675A",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "minWidth": 40,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="primary"
      fontSize={20}
      fontWeight={400}
      justification="center"
      padding={3}
      style={
        Array [
          Object {
            "color": "#33343E",
            "flexDirection": "row",
            "fontFamily": "Raleway",
            "fontSize": 20,
            "fontWeight": "400",
            "paddingBottom": 3,
            "paddingLeft": 3,
            "paddingRight": 3,
            "paddingTop": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`matches default snapshot 1`] = `
<View
  background="primary"
  opacity={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "rgba(51,52,62,1)",
        "borderColor": "#33343E",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "minWidth": 40,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="accent"
      fontSize={20}
      fontWeight={400}
      justification="center"
      padding={3}
      style={
        Array [
          Object {
            "color": "#FFD14A",
            "flexDirection": "row",
            "fontFamily": "Raleway",
            "fontSize": 20,
            "fontWeight": "400",
            "paddingBottom": 3,
            "paddingLeft": 3,
            "paddingRight": 3,
            "paddingTop": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`matches opacity snapshot 1`] = `
<View
  background="accent"
  border="red"
  opacity={0.3}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "rgba(255,209,74,0.3)",
        "borderColor": "#D4675A",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "minWidth": 40,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="primary"
      fontSize={20}
      fontWeight={400}
      justification="center"
      padding={3}
      style={
        Array [
          Object {
            "color": "#33343E",
            "flexDirection": "row",
            "fontFamily": "Raleway",
            "fontSize": 20,
            "fontWeight": "400",
            "paddingBottom": 3,
            "paddingLeft": 3,
            "paddingRight": 3,
            "paddingTop": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`matches text color snapshot 1`] = `
<View
  background="primary"
  opacity={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "rgba(51,52,62,1)",
        "borderColor": "#33343E",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "minWidth": 40,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      color="accent"
      fontSize={20}
      fontWeight={400}
      justification="center"
      padding={3}
      style={
        Array [
          Object {
            "color": "#FFD14A",
            "flexDirection": "row",
            "fontFamily": "Raleway",
            "fontSize": 20,
            "fontWeight": "400",
            "paddingBottom": 3,
            "paddingLeft": 3,
            "paddingRight": 3,
            "paddingTop": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      1
    </Text>
  </View>
</View>
`;
